build.gradle

```
@@ -53,10 +50,7 @@ java {
 wrapper {
     distributionType = 'ALL'
     doLast {
-        final DistributionLocator locator = new DistributionLocator()
-        final GradleVersion version = GradleVersion.version(wrapper.gradleVersion)
-        final URI distributionUri = locator.getDistributionFor(version, wrapper.distributionType.name().toLowerCase(Locale.ENGLISH))
-        final URI sha256Uri = new URI(distributionUri.toString() + ".sha256")
+        final URI sha256Uri = new URI(wrapper.getDistributionUrl() + ".sha256")
         final String sha256Sum = new String(sha256Uri.toURL().bytes)
         wrapper.getPropertiesFile() << "distributionSha256Sum=${sha256Sum}\n"
```

Since this is the wrapper task there is no need to access the wrapper task via the getter from the task container. This could in fact be seen as an issue by Gradle in the future, because a task is reaching out to another task that is within its closure.

```suggestion
        def sum = new String(new URL("${distributionUrl}.sha256").bytes)
        propertiesFile << "distributionSha256Sum=$sum"
```

---

Good catch. Fixed.